apps/rush-lib/src/cli/actions/RushCommandLineParser.ts

```diff
@@ -90,20 +90,29 @@ export default class RushCommandLineParser extends CommandLineParser {
   }
 
   private _execute(): void {
-    this.telemetry = new Telemetry(this.rushConfig);
+    if (this.rushConfig) {
+      this.telemetry = new Telemetry(this.rushConfig);
+    }
+
     super.onExecute();
-    this.flushTelemetry();
+
+    if (this.telemetry) {
+      this.flushTelemetry();
+    }
   }
 
   private _populateActions(): void {
     try {
-      this.rushConfig = RushConfiguration.loadFromDefaultLocation();
+      let  commandLineConfig: CommandLineConfiguration | undefined = undefined;
+
+      if (RushConfiguration.tryFindRushJsonLocation()) {
```

Comment:
Save the return value in a variable to avoid calling loadFromDefaultLocation to search again.

Reply:
Good idea

---
In reply to: [150166540](https://github.com/Microsoft/web-build-tools/pull/437#discussion_r150166540) [](ancestors = 150166540)